tests/v2/validation/provisioning/clustertemplates/provisioning_cluster_template_test.go

```diff
@@ -0,0 +1,335 @@
+	require.NoError(c.T(), err)
+
+	clusterTemplateRevision, err := rke1.CreateRkeTemplateRevision(c.client, clusterTemplateRevisionConfig, clusterTemplate.ID)
+	require.NoError(c.T(), err)
+
+	clusterTemplateClusterConfig := new(clustertemplates.ClusterTemplateConfig)
+	clusterTemplateClusterConfig.RKE1TemplateID = clusterTemplate.ID
+	clusterTemplateClusterConfig.RKE1TemplateRevisionID = clusterTemplateRevision.ID
+
+	if answers != nil {
+		clusterTemplateClusterConfig.RKETemplateAnswers = &mgmt.Answer{
+			Values: answers,
+		}
+	}
+	return clusterTemplateClusterConfig
+}
+
+func (c *ClusterTemplateProvisioningTestSuite) testCreateRke1ClusterWithTemplateAndVerify(client *rancher.Client, clusterTempClusterConfig *rke1.ClusterTemplateConfig) *mgmt.Cluster {
+
+	rke1Provider := provisioning.CreateRKE1Provider(c.provisioningConfig.NodeProviders[0])
+	nodeTemplate, err := rke1Provider.NodeTemplateFunc(client)
+	require.NoError(c.T(), err)
+
+	clusterConfig := clusters.ConvertConfigToClusterConfig(c.provisioningConfig)
+
+	nodeAndRoles := []provisioninginput.NodePools{
+		provisioninginput.AllRolesNodePool,
+	}
+	clusterConfig.NodePools = nodeAndRoles
+
+	clusterObject, err := provisioning.CreateProvisioningRKE1ClusterWithClusterTemplate(client, rke1Provider, clusterTempClusterConfig, clusterConfig, nodeTemplate)
+	require.NoError(c.T(), err)
+
+	clusterConfig.KubernetesVersion = c.provisioningConfig.RKE1KubernetesVersions[0]
+	provisioning.VerifyRKE1Cluster(c.T(), c.client, clusterConfig, clusterObject)
+
+	if clusterObject.RancherKubernetesEngineConfig.Services.Etcd.BackupConfig.S3BackupConfig != nil {
+		err := extensionsetcdsnapshot.CreateRKE1Snapshot(client, clusterObject.Name)
+		require.NoError(c.T(), err)
+
+		existingSnapshots, err := extensionsetcdsnapshot.GetRKE1Snapshots(client, clusterObject.ID)
+		etcdNodeCount, _ := etcdsnapshot.MatchNodeToAnyEtcdRole(client, clusterObject.ID)
+
+		_, err = provisioning.VerifySnapshots(client, clusterObject.Name, etcdNodeCount+len(existingSnapshots), true)
+		require.NoError(c.T(), err)
+	}
+
+	return clusterObject
+}
+
+func (c *ClusterTemplateProvisioningTestSuite) TestProvisioningRKE1ClusterWithClusterTemplate() {
+	log.Info("Create an rke template and creating a downstream node driver with the rke template.")
+
+	templateRevisionConfig := new(clustertemplates.ClusterTemplateRevisionConfig)
+	templateRevisionConfig.NetworkPlugin = c.provisioningConfig.CNIs[0]
+	templateRevisionConfig.RKE1KubernetesVersion = c.provisioningConfig.RKE1KubernetesVersions[0]
+
+	if c.provisioningConfig.ETCDRKE1 != nil {
+		templateRevisionConfig.BackupConfig = c.provisioningConfig.ETCDRKE1.BackupConfig
+	}
+
+	clusterConfig := c.createRke1ClusterTemplateAndTemplateRevisions(*templateRevisionConfig, nil, nil)
+
+	c.testCreateRke1ClusterWithTemplateAndVerify(c.client, clusterConfig)
+}
+
+func (c *ClusterTemplateProvisioningTestSuite) TestEnforceClusterTemplateProvisioningRKE1Cluster() {
```

Comment:
this may just be my own misunderstanding. But the test title sounds like I won't be able to provision using the normal 'cluster management -> create' page, but instead I would only be able to provision (for rke1) through cluster templates. Is that true? 
If so, can you add a check that provisioning directly through steve does not work (require.Error)
if not, can you clarify for me what the enforceClusterTemplate setting is supposed to do?

reading the code in this test, maybe it is supposed to only allow explicitly added users to be able to provision using the template? -> if this is the case, can you add a check that a new user can't provision using the template?

Reply:
this is a global setting and when its set, standard users cannot provision rke1 clusters without using the rke templates..I'm not sure what you mean but `create' page, but instead I would only be able to provision (for rke1) through cluster templates` but we can provision a cluster from cluster management page. 

Its just that the rke template is auto selected. Admins can still create clusters --> in this test we are verfying admins can still create clusters. For standard users, I have added a new task for which the aforementioned checks will be taken care. https://github.com/rancher/qa-tasks/issues/1550